include/mf_nh_operator.h

```diff
@@ -855,17 +857,17 @@ NeoHookOperator<dim, fe_degree, n_q_points_1d, Number>::do_operation_on_cell(
           case MFCaching::scalar_referential: {
             // Assert frame
             Assert(mf_frame == MFFrame::referential, ExcNotImplemented());
-            const VectorizedArrayType  one = make_vectorized_array<Number>(1.);
-            const VectorizedArrayType *cached_position =
-              &cached_second_scalar(cell, 0);
+            const VectorizedArrayType one = make_vectorized_array<Number>(1.);
+            VectorizedArrayType *     cached_position =
+              const_cast<VectorizedArrayType *>(&cached_second_scalar(cell, 0));
             constexpr unsigned int n_q_points =
               Utilities::pow(n_q_points_1d, dim);
 
             VectorizedArrayType *x_grads = phi.begin_gradients();
             // Only a data storage for the gradients is required, not a complete
             // copy of an FEEvaluation object. However, the copy constructor is
             // way faster than allocating an AlignedVector here.
-            FECellIntegrator     phi_grad(phi);
+            CellIntegrator       phi_grad(phi);
```

Comment:
TODO: replace cached quad_values by cached dof_values and evaluate gradients instead of the internal::collocation